### PR TITLE
Dedicated method for automation sidebar copies

### PR DESCRIPTION
**Pull request by piekczyk**
# [Dedicated method for automation sidebar copies](https://app.shortcut.com/oazo-apps/story/5774/dedicated-methods-for-handle-automation-sidebar-copies)

<please insert a shortcut link above>
  
## Changes 👷‍♀️
  <Please add short list of changes>

- prepared dedicated method for handling copies within automation features sidebars
- adjusted old methods which will stay for other features (removed automation flows handling from them)
  
## How to test 🧪
  <Please explain how to test your changes>

- test all flows within automation scope and pay attention to copies inside sidebar
- test stop loss setup during opening flow
- few test with opening / managing borrow and multiply vault to make sure that other copies are not broken

### CODE REVIEW

File: features/automation/common/AddAndRemoveTriggerControl.tsx [hunk 1]
```diff
@@ -65,7 +72,7 @@ export function AddAndRemoveTriggerControl({
     [removeAllowance, proxyAddress, triggersId],
   )
 
-  function txHandler() {
+  function txHandler(options?: AddAndRemoveTxHandler) {
```
Comment (piotrkonowrocki) [inline features/automation/common/AddAndRemoveTriggerControl.tsx:75]: Is there a reason to use some `options` var as proxy holding whole interface? Wouldn't be it easier just to use spread?
```
function txHandler({callOnSuccess}: AddAndRemoveTxHandler) {
```

File: features/automation/common/getAutomationPrimaryButtonLabel.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    case 'editBasicBuy':
+    case 'editConstantMultiple':
+      return 'automation.updating'
+    case 'cancelBasicSell':
+    case 'cancelBasicBuy':
+    case 'cancelConstantMultiple':
+    case 'cancelSl':
+      return 'automation.cancelling'
+    default:
+      return UNREACHABLE_CASE_MESSAGE
```
Comment (piotrkonowrocki) [inline features/automation/common/getAutomationPrimaryButtonLabel.ts:54]: Why does other methods are throwing exception, but here (and in method below) there is a empty string as default return?

I don't remember if it's somehow justified and it was like that already in older methods? If so, I suggest moving `UNREACHABLE_CASE_MESSAGE` to 'helpers/UnreachableCaseError' and importing it in similar way like `UnreachableCaseError`.

Comment (piotrkonowrocki) [inline features/automation/common/getAutomationPrimaryButtonLabel.ts:54]: Why does other methods are throwing exception, but here there is a empty string as default return?

I don't remember if it's somehow justified and it was like that already in older methods? If so, I suggest moving `UNREACHABLE_CASE_MESSAGE` to 'helpers/UnreachableCaseError' and importing it in similar way like `UnreachableCaseError`.

Same goes for line `72` below.

File: features/sidebar/getSidebarStatus.ts [hunk 3]
```diff
@@ -18,25 +18,25 @@ function getSidebarProgressTxInProgressKey({
...
-      return 'auto-buy.cancelling-auto-buy'
-    case 'addBasicBuy':
-    case 'editBasicBuy':
-      return 'auto-buy.setting-auto-buy'
-    case 'cancelConstantMultiple':
-      return 'constant-multiple.cancelling-constant-multiple'
-    case 'addConstantMultiple':
-    case 'editConstantMultiple':
-      return 'constant-multiple.setting-constant-multiple'
+    // case 'adjustSl':
```
Comment (piotrkonowrocki) [inline features/sidebar/getSidebarStatus.ts:22]: Why commented out instead of removing? Same for `52` line in the same file.

**Review by piotrkonowrocki — CHANGES_REQUESTED** [review]

File: features/automation/common/AddAndRemoveTriggerControl.tsx (hunk 1 above)
Comment (piekczyk) [reply features/automation/common/AddAndRemoveTriggerControl.tsx:75]: I tried to do it this way but typescript does not allow for object destructuring when parameter is optional

**Review by piotrkonowrocki — APPROVED** [review]

**Pull request merged**